samples/rust/rust_selftests.rs

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Self test cases for Rust.
+
+use kernel::prelude::*;
+// Keep the `use` for a test in its test function. Module-level `use`s are only for the test
+// framework.
+
+module! {
+    type: RustSelftests,
+    name: b"rust_selftests",
+    author: b"Rust for Linux Contributors",
+    description: b"Self test cases for Rust",
+    license: b"GPL",
+}
+
+struct RustSelftests;
+
+/// A summary of testing.
+///
+/// A test can
+///
+/// * pass (successfully), or
+/// * fail (without hitting any error), or
+/// * hit an error (interrupted).
+///
+/// This is the type that differentiates the first two (pass and fail) cases.
+///
+/// When a test hit an error, the test function should skip and return the error. Note that this
+/// doesn't mean the test fails, for example if the system doesn't have the enough memory for
+/// testing, the test function may return a `ENOMEM` and skip.
```

Comment:
```suggestion
/// When a test hits an error, the test function should skip and return the error. Note that this
/// doesn't mean the test fails, for example if the system doesn't have enough memory for
/// testing, the test function may return an `Err(ENOMEM)` and skip.
```

Reply:
Sorry, my bad, I edited it and then realized I was right the first time, no need for the `Error::`. :)

Reply:
No worries, let me get that fixed.